### PR TITLE
Changeling Meteor event now breaks more gracefully and will absolutely make sure the changeling gets where it should

### LINKED ISSUE #73349: Meteor Changelings can get immediately killed by shield satellites
Reporting client version: 514.1589

## Round ID:
[199870](https://scrubby.melonmesa.com/round/199870)

## Reproduction:
This just seems like a waste of threat, you are able to spawn in and start flying towards the station but it just immediately kills you upon you closing in. I feel like there should be SOME sort of counter to being instantly round removed upon spawning in.

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

The changeling meteor now has makes sure that, in the event of a near miss, meteor shield interception, or even a miner swatting it midair with a pickaxe, it's stored changeling will reach the station.

If the meteor "dies", it will initiate an emergency eject sequence and launch the changeling through space towards their destination. If intercepted by a shield satellite, the stored changeling will be thrown into the shield instead (because it's closer).

Is it silly? Yeah. Does it look dumb and put our fragile immersion at risk? That too. Unfortunately, I really don't have any better ideas beyond teleporting them to a nearstation turf and saying "yeah man the bluespace did it".

Some code related to the meteor shields and how meteors shot at has been shuffled around to better accommodate this. This may or may not function as a framework for customizing how individual meteor types interact with meteor shields.

Also, the changeling meteor no longer gives an achievement when you examine it. You spawn INSIDE of the meteor after all, so examining it shouldn't be cause for celebration.

## Why It's Good For The Game

Closes #73349.

Changelings getting absolutely vaporized before they can make landfall (or missing the station entirely) is really funny, but it makes work for the admins and takes away threat from the round (boring!).

## Changelog
:cl:
fix: Midround Changelings are now ejected from their meteor when something bad happens, ensuring they reach the station.
/:cl:

### CODE REVIEW

File: code/modules/events/ghost_role/changeling_event.dm [hunk 1]
```diff
@@ -98,3 +105,22 @@
 
 /obj/effect/meteor/meaty/changeling/ram_turf()
 	return //So we don't instantly smash into our occupant upon unloading them.
+
+/obj/effect/meteor/meaty/changeling/shield_defense(obj/machinery/satellite/meteor_shield/defender)
+	landing_target = defender
+	return TRUE
+
+/obj/effect/meteor/meaty/changeling/handle_stopping() //If the meteor misses the station and deletes itself, we make absolutely sure the changeling reaches the station.
```
Comment (san7890) [inline code/modules/events/ghost_role/changeling_event.dm:113]:
```suggestion
// If the meteor misses the station and deletes itself, we make absolutely sure the changeling reaches the station.
/obj/effect/meteor/meaty/changeling/handle_stopping()
```

better for people with smol screens

File: code/modules/events/ghost_role/changeling_event.dm [hunk 2]
```diff
@@ -98,3 +105,22 @@
...
+	if(!landing_target)
+		landing_target = locate(/obj/effect/landmark/observer_start) in GLOB.landmarks_list //If our destination turf is gone for some reason, we chuck them at the observer_start landmark (usually at the center of the station) as a last resort.
```
Comment (san7890) [inline code/modules/events/ghost_role/changeling_event.dm:115]:
```suggestion
		//If our destination turf is gone for some reason, we chuck them at the observer_start landmark (usually at the center of the station) as a last resort.
		landing_target = locate(/obj/effect/landmark/observer_start) in GLOB.landmarks_list
```

looks cleaner IMO

File: code/modules/events/ghost_role/changeling_event.dm [hunk 3]
```diff
@@ -98,3 +105,22 @@
...
+
+	for(var/atom/movable/changeling in contents)
+		changeling.forceMove(get_turf(src))
+		changeling.throw_at(landing_target, 2, 2)
+		changeling.visible_message(span_warning("[changeling] is launched out from inside of the [name]"), span_changeling("Sensing that something is terribly wrong, we forcibly eject ourselves from the [name]!"))
+		playsound(changeling, 'sound/effects/splat.ogg', 50, pressure_affected = FALSE)
+
+	..()
```
Comment (san7890) [inline code/modules/events/ghost_role/changeling_event.dm:123]:
```suggestion
	return ..()
```

style nit

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/modules/events/ghost_role/changeling_event.dm (hunk 2 above)
Comment (san7890) [reply code/modules/events/ghost_role/changeling_event.dm:115]: also, do you know if this works on multi-z stations? What happens if the observer_start landmark is on the top-z, but our meteor is on the bottom-z? can we account for that?

**Comment by san7890** [comment]
> Also, the changeling meteor no longer gives an achievement when you examine it. You spawn INSIDE of the meteor after all, so examining it shouldn't be cause for celebration.

I didn't see this code anywhere?

File: code/modules/events/ghost_role/changeling_event.dm (hunk 2 above)
Comment (san7890) [reply code/modules/events/ghost_role/changeling_event.dm]: honestly might be worth to ensure that we only ever spawn on the z-level of the observer-start landmark just to ensure that this diversion works? not sure at all.

**Comment by jlsnow301** [comment]
get there safe lil fella

**Pull request merged**